src/vm.ts

```diff
@@ -218,6 +225,15 @@ function step(n: number = 0): void {
   let resValue: string | number = runSheet
     .getRange(`${resColumn}${n + 2}`)
     .getDisplayValue();
+  
+  let operandsValue: operands = {
+    op0: op0Value,
+    op1: op1Value,
+    res: resValue,
+    dst: dstValue,
+  };
+
+  opcodeAssertions(instruction, operandsValue);
```

Comment:
I'd rather put the checks right after the undefined computation, ie in the `switch (instruction.Opcode)` like

```
  switch (instruction.Opcode) {
    case Opcodes.Call:
      // TODO: update to be like for Opcodes.AssertEq and write values only if it's undefined
      runSheet
        .getRange(op0Addr)
        .setValue(registers[Registers.PC] + size(instruction));
      runSheet.getRange(dstAddr).setValue(registers[Registers.FP]);
      // TODO: add assertion here
      break;
    case Opcodes.AssertEq:
      switch (instruction.ResLogic) {
      // ...
      }
      // TODO: add assertion here
```